source/pdo_sqlsrv/pdo_stmt.cpp

```diff
@@ -1067,6 +1067,14 @@ int pdo_sqlsrv_stmt_next_rowset( _Inout_ pdo_stmt_t *stmt TSRMLS_DC )
 
         SQLSRV_ASSERT( driver_stmt != NULL, "pdo_sqlsrv_stmt_next_rowset: driver_data object was null" );
 
+        // If SQLNumResultCols returns 0, the result set is empty. Normally this error is 
+        // handled in core_sqlsrv_fetch, but if the user calls nextRowset() before fetch()
+        // the error message won't show up unless we handle it here.
+        SQLSMALLINT has_fields = core::SQLNumResultCols( driver_stmt TSRMLS_CC );
+        CHECK_CUSTOM_ERROR( has_fields == 0, driver_stmt, SQLSRV_ERROR_NO_FIELDS ) {
+            throw core::CoreException();
+        }
+
```

Comment:
From this [page](https://docs.microsoft.com/en-us/sql/relational-databases/native-client-odbc-api/sqlnumresultcols) it seems valid and accurate to call this without making another trip to the server. Please confirm.

Reply:
Since a statement has to be executed for the user to call nextRowset, there is no trip to the server made here.

Reply:
Yes, I've verified the same problem exists on the sqlsrv side. But I think core_sqlsrv_fetch is never called if nextRowset is called before fetch, which is why we're catching the error here. I'll add a fix and test for sqlsrv.